### DISCUSSION TITLE
Offer cloudflared run-parameters as add-on option

### DISCUSSION THREAD

**Discussion by brenner-tobias** [homeassistant-apps/app-cloudflared#572] [Ideas]
Referring to this [raised issue](https://github.com/brenner-tobias/ha-addons/issues/19), I would like to follow up on the question if we should offer [protocol](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/configure-tunnels/tunnel-run-parameters/#protocol) as option for our add-on.

I did initially refuse the suggestion, but learned that the protocol can indeed only be changed on the tunnel client side.

Now building on this, there are a couple of [run parameters](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/configure-tunnels/tunnel-run-parameters/) that can only be added on the client. 

Since there is no way of knowing which ones someone need (for experienced users only obviously), my suggestion is to add an optional string "run-parameters" which is then added to the start-up string of Cloudflared. That way, any and also multiple parameters can be added by the users.

Kindly let me know if you have any questions, I am looking forward to your perspective.

**Comment by brenner-tobias** [comment]
@elcajon

**Comment by elcajon** [comment]
I think that's a good suggestion. An option (perhaps as a list) to add further parameters makes perfect sense in my opinion.

**Reply by brenner-tobias** [reply]
Not sure how a list would work, since there are different parameters with different options, so this would not work, or am I missing something?
Since this is a feature for experienced users, I would simply offer a string that gets added to the run command

**Reply by elcajon** [reply]
Yes, you're right, let's just start with a `str?`

**Reply by baflo** [reply]
First of all, thank you for dealing with my problem 👍

I'd be fine with the `str?` option for the tunnel parameter, but another way would be to allow arbitrary environment variables to be set in the container, as all the [run parameter](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/configure-tunnels/tunnel-run-parameters/) know a corresponding env var. Maybe that is more versatile.

Actually, that's the way I tested it for me: I forked your repo and changed the option `environment` in the `config.yaml`.

**Reply by brenner-tobias** [reply]
Thanks for the suggestion, but I do not get what you mean exactly. Can you give me some more detail how you would offer the env variable as an add-on option and where you would set it?
Also, feel free to raise a PR to discuss further

**Comment by baflo** [comment]
I thought you could just offer an "object" option to map env vars which would be exported in the container by the entrypoint script. But I just saw, that the addon configuration doesn't allow object options:
![image](https://github.com/brenner-tobias/addon-cloudflared/assets/834350/399c676f-e452-4fdc-b2b6-953f1c070c44)

**Comment by elcajon** [comment]
Thinking about the possibility to add the `--logfile` parameter with the latest changes... We currently don't have any Home Assistant folders mapped into the Cloudflared container, which makes the option not very useful at the moment. There have been some changes to the available [folder mapping options](https://developers.home-assistant.io/blog/2023/11/06/public-addon-config). Maybe we should add the `addon_config` option for this?

**Reply by brenner-tobias** [reply]
agreed, let's add the folder

**Reply by elcajon** [reply]
#578

**Reply by elcajon** [reply]
Just a thought... using this mapping, we also have the option of reading another (.env) file from this directory, which could process environment variables not provided in the add-on options.

**Reply by brenner-tobias** [reply]
What kind of variables would you define there that cannot be added by the current method with the array?

**Reply by elcajon** [reply]
I don't have a concrete idea at the moment. Just thought we could keep it in mind for further requirements.

**Reply by brenner-tobias** [reply]
got it, good point
